.github/workflows/scripts/add-labels.rb

```diff
@@ -0,0 +1,27 @@
+require "octokit"
+
+client = Octokit::Client.new(access_token: ENV["GITHUB_TOKEN"])
+
+repo = ENV["GITHUB_REPOSITORY"]
+pr_number = ENV["GITHUB_REF"].split("/")[2]
+
+pr_files = client.pull_request_files(repo, pr_number)
+
+pr_labels = {
+  no_bottles: 0
+}
+
+pr_files.each do |file|
+  formula = file[:filename]
+
+  if File.exist?(formula) && File.read(formula).match(/bottle :unneeded/)
+    pr_labels[:no_bottles] += 1
+  end
+end
+
+if pr_files.count == pr_labels[:no_bottles]
```

Comment:
Will this work as expected if we have deleted files/formulae in a PR? And the same question about the case when we update a formula which has a versioned analogue (when PR contains alias changing)?

Reply:
It doesn't add labels if a formula has been deleted, thanks to the `File.exist?` check.

Reply:
However, you raise a good point about the aliases, that was one thing I didn't try [over in my testing repo](https://github.com/issyl0/homebrew-actions-testing/). I'll give it a go and push a fix if it doesn't work!

Reply:
This action will [only trigger on changes to `Formula/*.rb`](https://github.com/Homebrew/homebrew-core/pull/48911/files#diff-b59ac451da8b4fc7c34118b14fb4d067R5-R6), so I think we'll be fine for aliases? What behaviour were you expecting?

Reply:
I'm talking about PRs like https://github.com/Homebrew/homebrew-core/pull/48823 (but with `bottle :unneeded`), I'm not sure, but `pr_files.count` will be 2 in this case, but `pr_labels[:no_bottles]` is only 1 (I haven't checked it, just assuming)

Sorry, I poorly formulated the question 🙂

Reply:
Thanks - I understand the problem now! I've fixed this in #48919.